File: Engine/Scripts/Rotator.lua
Rotator = {}

function Rotator:Create()


end

function Rotator:GatherProperties()

    return 
    {
        { name = "angularVelocity", type = DatumType.Vector }
    }

end

function Rotator:Tick(deltaTime)

    self:AddRotation(self.angularVelocity * deltaTime)
    
end

File: Tools/PolyphaseSharp/Tests/Sample/Scripts/CSharp/Rotator.cs
using Polyphase;

public class Rotator : Script3D
{
    [Property] public Vector3 AngularVelocity = new Vector3(0, 90, 0);
    [Property(Display = "Spin Enabled")] public bool Enabled = true;

    private float mElapsed;

    public override void Start()
    {
        Log.Debug("Rotator started on " + Name);
    }

    public override void Tick(float deltaTime)
    {
        if (!Enabled) { return; }
        mElapsed += deltaTime;
        AddRotation(AngularVelocity * deltaTime);
    }
}
